File: java/src/main/java/org/mydotey/scf/apollo/ApolloConfigurationSource.java
package org.mydotey.scf.apollo;

import org.mydotey.scf.source.stringproperty.StringPropertyConfigurationSource;

/**
 * @author koqizhao
 *
 * Jul 24, 2018
 */
public class ApolloConfigurationSource extends StringPropertyConfigurationSource<ApolloConfigurationSourceConfig> {

    public ApolloConfigurationSource(ApolloConfigurationSourceConfig config) {
        super(config);
        getConfig().getApolloConfig().addChangeListener(e -> ApolloConfigurationSource.this.raiseChangeEvent());
    }

    @Override
    public String getPropertyValue(String key) {
        return getConfig().getApolloConfig().getProperty(key, null);
    }

}

File: dotnet/src/MyDotey.SCF.Apollo/ApolloConfigurationSource.cs
using System;

using Com.Ctrip.Framework.Apollo;

using MyDotey.SCF.Source.StringProperty;

namespace MyDotey.SCF.Apollo
{
    /**
     * @author koqizhao
     *
     * Jul 24, 2018
     */
    public class ApolloConfigurationSource : StringPropertyConfigurationSource<ApolloConfigurationSourceConfig>
    {
        public ApolloConfigurationSource(ApolloConfigurationSourceConfig config)
            : base(config)
        {
            Config.ApolloConfig.ConfigChanged += (o, e) => RaiseChangeEvent();
        }

        public override string GetPropertyValue(string key)
        {
            return Config.ApolloConfig.GetProperty(key, null);
        }
    }
}